autonomy/cli/scaffold_fsm.py

```diff
@@ -36,24 +36,53 @@
 from aea.configurations.constants import SKILL
 
 # the decoration does side-effect on the 'aea scaffold' command
-from aea.configurations.data_types import PublicId
+from aea.configurations.data_types import PackageId, PackageType, PublicId
+from aea.package_manager.v1 import PackageManagerV1
 
 from autonomy.constants import ABSTRACT_ROUND_ABCI_SKILL_WITH_HASH
-from autonomy.fsm.scaffold.scaffold_skill import ScaffoldABCISkill
+from autonomy.fsm.scaffold.scaffold_skill import (
+    ScaffoldABCISkill,
+    SkillConfigUpdater,
+    TO_LOCAL_REGISTRY_FLAG,
+)
 
 
 def _add_abstract_round_abci_if_not_present(ctx: Context) -> None:
     """Add 'abstract_round_abci' skill if not present."""
-    abstract_round_abci_public_id = PublicId.from_str(
-        ABSTRACT_ROUND_ABCI_SKILL_WITH_HASH
+    MSG = (
+        "Skill valory/abstract_round_abci not found in agent dependencies, adding it..."
     )
-    if abstract_round_abci_public_id.to_latest() not in {
-        public_id.to_latest() for public_id in ctx.agent_config.skills
-    }:
-        click.echo(
-            "Skill valory/abstract_round_abci not found in agent dependencies, adding it..."
+
+    if ctx.config.get(TO_LOCAL_REGISTRY_FLAG):
+        skill_id = SkillConfigUpdater.get_actual_abstract_round_abci_package_public_id(
+            ctx
+        )
+
+        if not skill_id:
+            # add package to local registry
+            click.echo(MSG)
+            package_manager = PackageManagerV1.from_dir(Path(ctx.registry_path))
+            package_id = PackageId(
+                package_type=PackageType.SKILL,
+                public_id=PublicId.from_str(ABSTRACT_ROUND_ABCI_SKILL_WITH_HASH),
+            )
+            package_manager.add_package(package_id)
```

Comment:
You should use `add_item` method here because `BasePackageManager.add_package` does not support recursive fetching for missing packages.